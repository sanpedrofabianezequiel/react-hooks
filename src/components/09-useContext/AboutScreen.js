import React, { useContext } from 'react'
import { UserContext } from './UseContext';

export const AboutScreen = () => {

    const {user,setUser}= useContext(UserContext);

    const handleClick=()=>{
        setUser({});
    };

    return (
        <div>
            <h3>AboutScreen</h3>
            <hr/>

            <pre>
                {JSON.stringify(user,null,3)}
            </pre>

            <button className="btn btn-warning" onClick={handleClick}>
                Limpiar
            </button>
        </div>
    )
}
