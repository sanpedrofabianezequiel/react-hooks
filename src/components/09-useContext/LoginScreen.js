import React, { useContext } from 'react'
import { UserContext } from './UseContext';

export const LoginScreen = () => {

    const {user,setUser}= useContext(UserContext);

    const newUser={
        id:123,
        name:'noelia'
    }
    return (
        <div>
            <h3>LoginScreen</h3>
            <hr/>

            <button className="btn btn-primary" onClick={ () => setUser(newUser)}>
                Login
            </button>
        </div>
    )
}
