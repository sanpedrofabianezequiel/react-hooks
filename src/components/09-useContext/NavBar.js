import React from 'react'
import {Link, NavLink} from 'react-router-dom';


export const NavBar = () => {
    /*return (
        <nav>
            <ul>
                <li>
                    <Link to="/">Home</Link>
                </li>
                <li>
                    <Link to="/about">About</Link>
                </li>
                <li>
                    <Link to="/login">Login</Link>
                </li>
            </ul>
        </nav>
    )*//*
    return(
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link to="/" className="navbar-brand" >useContext  </Link>
            <div className="collapse navbar-collapse" id="navbarNav">
                <div className="navbar-nav">
                        <Link to="/" className="nav-item nav-link active" >Home</Link>
                        <Link to="/about" className="nav-item nav-link" >About</Link>
                        <Link to="/login" className="nav-item nav-link" >Login</Link>
                </div>
            </div>
        </nav>
    );*/
    return(
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <NavLink  exact={true} activeClassName="active" to="/use" className="navbar-brand" >useContext  </NavLink>
            <div className="collapse navbar-collapse" id="navbarNav">
                <div className="navbar-nav">
                        <NavLink exact={true} activeClassName="active" to="/" className="nav-item nav-link " >Home</NavLink>
                        <NavLink exact={true} activeClassName="active" to="/about" className="nav-item nav-link" >About</NavLink>
                        <NavLink exact={true} activeClassName="active" to="/login" className="nav-item nav-link" >Login</NavLink>
                </div>
            </div>
        </nav>
    );

}
