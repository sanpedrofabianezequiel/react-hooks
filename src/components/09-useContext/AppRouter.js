import React from 'react'


import {HomeScreen} from './HomeScreen';
import {AboutScreen} from './AboutScreen';
import {LoginScreen} from './LoginScreen';
import{
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from 'react-router-dom'
import { NavBar } from './NavBar';



export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className="container">

                <Switch>
                    <Route exact={true} path="/" component={HomeScreen} />
                    
                    <Route exact={true} path="/about" component={AboutScreen} />
                    
                    <Route exact={true} path="/login" component={LoginScreen} />

                    <Redirect to="/" />
                </Switch>
                </div>
            </div>
        </Router>
    )
}
