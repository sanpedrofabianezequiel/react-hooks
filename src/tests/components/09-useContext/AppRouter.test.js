import React from 'react';
import {shallow,mount} from 'enzyme';
import '@testing-library/jest-dom';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UseContext';

describe('Pruebas en <AppRouter/>', () => {
    
    const user =  {
        id:123,
        email:"user@email.com"
    }

    const wrapper =  shallow(
    
        <UserContext.Provider value={
            {user:user}
        }>
            <AppRouter/>
        </UserContext.Provider>
        
    );
    test('de de mostrarse el correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
