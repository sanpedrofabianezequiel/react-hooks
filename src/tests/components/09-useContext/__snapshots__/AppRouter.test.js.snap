// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> de de mostrarse el correctamente 1`] = `<AppRouter />`;
